### DISCUSSION TITLE
Directory preference order

### DISCUSSION THREAD

**Discussion by oleg-derevenetz** [ihhub/fheroes2#3631] [Q&A] [answered]
Hi @ihhub do you happen to know, why in `Settings::GetRootDirs()` game directory have a preference over user's home directory? Shouldn't it be the other way around? For example, `fheroes2.cfg` is saved into user's directory, but if there is another `fheroes2.cfg` in game directory, it will always be read instead of the one in user's directory. As a result, the settings are constantly getting lost.

**Comment by ihhub** [comment, marked as answer]
I think it's related to #3636 . As we discussed if the directory where the game locates in writeable then we save in it, otherwise in user's home directory. Reading and writing must be done from the same location. This was decided for data / maps files. Regarding configuration file - yes, user directory should be first.
